src/toast/toast.js

```diff
@@ -63,31 +64,35 @@ class Toast extends React.Component<ToastPropsT, ToastPrivateStateT> {
   }
 
   clearTimeout() {
-    [this.autoHideTimeout, this.animateOutCompleteTimer].forEach(timerId => {
+    [
+      this.autoHideTimeout,
+      this.animateInTimer,
+      this.animateOutCompleteTimer,
+    ].forEach(timerId => {
       if (timerId) {
         clearTimeout(timerId);
       }
     });
   }
 
   animateIn = () => {
-    this.setState({isHidden: false, isAnimating: true});
+    // Defer to next event loop
+    this.animateInTimer = setTimeout(() => {
+      this.setState({isVisible: true});
+    }, 0);
```

Comment:
Strangely this wasn't working without this setTimeout, even though animateIn should only be called after component is already mounting.

Reply:
animate in and animate out do pretty much the same, except operate with opposite values of state. Should we combine them into one animate function and even remove one of timers ` this.animateInTimer, this.animateOutCompleteTimer` ?